Source/Dafny/Translator.cs

```diff
@@ -14381,12 +14380,8 @@ public Expression GetSubstitutedBody(LetExpr e)
                   Bpl.Expr obj = TrExpr(e.Obj);
                   args.Add(obj);
                   result = new Bpl.NAryExpr(expr.tok, new Bpl.FunctionCall(translator.GetReadonlyField(field)), args);
-                  isLit = translator.IsLit(obj);
                 }
                 result = translator.CondApplyUnbox(expr.tok, result, field.Type, expr.Type);
-                if (isLit) {
-                  result = MaybeLit(result, translator.TrType(expr.Type));
-                }
```

Comment:
If I understand correctly, this change results in fewer `Lit`s. Does that mean there will be cases where the SMT solver could compute thanks to `Lit`s before but now cannot any more?

Reply:
Yes, you are right, that's a good point. I added a test case to illustrate the difference in expressivity.
The reason that Rustan pointed out to me why we still want to not have `Lit`s around `const`s (and not, for example, check whether the right-hand side of a `const` assignment has a `Lit`) is that suppose we have
```ml
module Const {
  export provides N
  const N := 3;
}

module Test {
import C = Const

  lemma test() {
    assert C.N == 3; // will fail
  }
}
```
The assertion will fail because the definition of `N` is hidden inside the `Const` module. If `N` were wrapped inside a `Lit` then in the outside module we might try to evaluate expressions that use `N` which we want to prevent. Therefore, we don't want `const`s to be `Lit`-wrapped at all.